src/final/ArduCAM.cpp

```diff
@@ -124,11 +124,28 @@ ArduCAM::ArduCAM() {
 
 void ArduCAM::begin() {
 	// [TODO] set up CS GPIO pin, (unnecessary if spi is configured with hardware slave management)
+	
+	// set to alternate function mode (10)
+	GPIOB->MODER &= ~(GPIO_MODER_MODER12_0); // PB12
+	//set AF to pin PB12
+	GPIOB->AFR[1] &= ~(GPIO_AFRH_AFSEL12);
```

Comment:
```suggestion
	// set to general purpose output mode (01)
	GPIOB->MODER |= GPIO_MODER_MODER12_0;
```

If you're gonna do software managed NSS, you should set the mode as regular GPIO output (exactly like setting up the LED).